STAIRCTL.md

```diff
@@ -0,0 +1,151 @@
+
+```
+Built-In Commands
+        clear: Clear the shell screen.
+        exit, quit: Exit the shell.
+        help: Display help about available commands.
+        history: Display or save the history of previously run commands
+        script: Read and execute commands from a file.
+        stacktrace: Display the full stacktrace of the last error.
+
+Connect Commands
+        connect: Connect to a Stairway database
+        disconnect: Disconnect from a Stairway database
+        show connection: Show the current connection
+
+Flight Commands
+        count flights: Count flights
+        count owned: Count owned flights
+        force fatal: Force a flight to FATAL state (dismal failure)
+        force ready: Force a flight to the READY state (disown it)
+        get flight: Get one flight
+        list dismal: List dismal failure flights
+        list flights: List flights
+        list owned: List owned flights
+
+Stairway Commands
+        list stairways: List stairway instances
+```
+
+## Connecting to a Stairway
+The most complex part of using StairCtl is connecting to a Stairway; that is, providing the
+information for connecting to a Stairway database.
+
+### `connect` command
+The full syntax or the `connect` command is:
+```
+connect [--username string]  [--password string]  [--dbname string]  [--host string]  [--port string]  
+```
+
+This table shows the precedence and defaults for each part of the input. Switches override
+environment variables, which override the default. 
+
+| **Option** | **Environment Variable** | **Default** |
+|:---|:---|:---|
+| `-u, -U, --username`| STAIRCTL_USERNAME | stairwayuser |
+| `-w, --password` | STAIRCTL_PASSWORD | starwaypw |
+| `-h, --host` | STAIRCTL_HOST | 127.0.0.1 |
+| `-p, --port` | STAIRCTL_PORT | 5432 |
+| `-d, --dbname` | STAIRCTL_DBNAME | stairwaylib |
+
+### `disconnect` command
+You can disconnect from a Stairway with the `disconnect` command. It takes no options.
+Connecting to a new Stairway database will automatically disconnect you from the current one.
+
+### `show connection` command
+Shows information for the current connection. It does not display the password.
+
+## Querying Flights
+One of the initial goals for StairCtl is to make it easier to look at flights, see what
+their state is, and see how they were executed.
+
+### `list` and `count` commands
+There are three list commands. They each provide **paging** of the list. You can specify
+`-o` or `--offset` for the start of the page and `-l` or `--limit` for the number of flights
+to list.
+
+* `list flights` command provides an optional `-s` or `--status` option to filter for flights in
+specific states. For example, `list flights --status QUEUED`.
+
+* `list dismal` command is a shortcut for `list flights --status FATAL`; providing a list
+of flights that require manual intervention for recovery.
+
+* `list owned` command shows flights that are currently owned by a Stairway instance. Those might
```

Comment:
not necessarily this PR, but is there a way to filter to flights that are not owned? or flights owned by a particular instance? Probably not as important to start with.

Reply:
Sounds good. I added those to the futures list.